solr/core/src/java/org/apache/solr/servlet/RequestRateLimiter.java

```diff
@@ -152,14 +143,22 @@ public RateLimiterConfig(SolrRequest.SolrRequestType requestType, boolean isEnab
     }
   }
 
-  // Represents the metadata for an acquired slot
-  static class AcquiredSlotMetadata {
-    public RequestRateLimiter requestRateLimiter;
-    public Semaphore usedPool;
+  // Represents the metadata for a slot
+  static class SlotMetadata {
+    private Semaphore usedPool;
 
-    public AcquiredSlotMetadata(RequestRateLimiter requestRateLimiter, Semaphore usedPool) {
-      this.requestRateLimiter = requestRateLimiter;
+    public SlotMetadata(Semaphore usedPool) {
       this.usedPool = usedPool;
     }
+
+    public void decrementRequest() {
+      if (usedPool != null) {
+        usedPool.release();
+      }
+    }
+
+    public boolean isUsedPoolNull() {
```

Comment:
This seems like an implementation detail that we don't need to expose?

Reply:
We need this to avoid caching a request when request rate limiter is disabled. Renaming the method.